doc/Requirements.md

```diff
@@ -0,0 +1,374 @@
+
+A syncer MUST support setting the mapped parameters value to the mapped hardware.
+(Why: to synchronise hardware on parameter change.)
+
+Syncers MUST retrieve and set the parameters value from the PF core.
+(Why: to be able to synchronise with hardware.)
+
+This API MAY be a packed parameter structure, following the C abi without padding.
+(Note: This is what the reference implementation does.)
+(Why: ???)
+
+## Introspection
+The syncer API SHOULD allow introspection of the mapped parameters.
+(Why: the parameter structure may be useful for the syncer to communicate with the hardware.)
+
+## Plugins
+The PF MUST be able to create syncers.
+(Why: to bind on the corresponding parameters.)
+
+### Definition
+The PF creates syncer using syncer builder.
+
+### Identifier
+All syncer mapping the same hardware SHOULD have their builder regrouped in a syncer library.
+(Why: to be able to link a group of parameters and a given hardware.)
+
+A syncer builder MUST have a unique identifier in its containing syncer library.
+(Why: To uniquely identify the syncer that should bind on parameters.
+      Given that the syncer library has already been specified.)
+
+A syncer library MUST have a unique identifier in the host system.
+(Why: To identify the library associated to parameters.)
+
+Syncer build and syncer library identifiers SHOULD be strings.
+(Why: The reference application does so.)
+
+### Loading
+Syncer library or/and builder MAY be loaded from dynamicly linked libraries (called syncer plugins).
+(Why: The reference implementation supports it.)
+
+Such syncer plugins SHOULD have an unique entry point that -when called- should register
+its payload (syncer library/builder) in the provided gatherer.
+(Why: The reference implementation supports it.)
+
+Multiple syncer plugins, may depend on each other. The PF should appropriatly handle the case
+and not fail.
+(Why: The reference implementation supports it.)
+
+## Mapping
+### Definition
+A parameter not binded to a syncer is called a "virtual parameter".
+(Todo: remove if not used in the requirements.)
+
+### Requirements
+**TODO**:
+ - Plugins
+ - association builder <-> parameters
+
+# Rule based dynamic abstraction
+
+## Philosophy
+
+The PF offers parameters mapped on hardware. This is a good but week abstraction.
+There is often a 1/1 relation between a parameter and the hardware it maps.
+
+A PF offers a mechanism to abstract the parameters to more high level concept.
+
+The goal is to hide numerous parameters and their dynamic value behind simple
+and human friendly api.
+
+It works by regrouping parameter with similar management and defining configurations
+for each "scenario". These "scenario" are then given a priority and a detection predicate.
+Configuration are then applied when their associated "scenario" is detected.
+
+"Scenario" are detected through arbitrary criterion provided by the PF host.
+
+## Definition
+
+Configurations: set of values for different parameters.
```

Comment:
I prefer my definition because configuration could be used without domains whereas in yours they are bound.
~~~
Each configuration SHOULD be in a "domain" (see Definition chapter).
(Why: Domains are mostly a way to define the priority of "Scenarios" for some parameters.
      It is not a MUST because this goal could also be achieve with (for example) global configurations and per parameter priority.
      It is not a MAY because the reference implementation uses domains.)
~~~